evaluate.py

```diff
@@ -160,13 +160,140 @@
+                create temp table _mig_eval_t (
+                    n int, n_tup_ins numeric, n_tup_upd numeric, n_tup_del numeric, xact_commit numeric
+                );
+
+                begin;
+                    insert into _mig_eval_t
+                        select 1, sum(n_tup_ins) n_tup_ins, sum(n_tup_upd) n_tup_upd, sum(n_tup_del) n_tup_del, xact_commit
+                        from pg_stat_user_tables u join pg_stat_database d on true
+                        where
+                            u.relname not in ('_mig_eval_t') AND
+                            u.schemaname not in (
+                                '_timescaledb_config', '_timescaledb_catalog', '_timescaledb_cache',
+                                'timescaledb_experimental', 'timescaledb_information', '_timescaledb_functions',
+                                'information_schema', 'pg_catalog') AND
+                            d.datname = current_database()
+                        group by xact_commit;
+                end;
+
+                select pg_sleep(@wait@);
+
+                begin;
+                    insert into _mig_eval_t
+                        select 2, sum(n_tup_ins) n_tup_ins, sum(n_tup_upd) n_tup_upd, sum(n_tup_del) n_tup_del, xact_commit
+                        from pg_stat_user_tables u join pg_stat_database d on true
+                        where
+                            u.relname not in ('_mig_eval_t') AND
+                            u.schemaname not in (
+                                '_timescaledb_config', '_timescaledb_catalog', '_timescaledb_cache',
+                                'timescaledb_experimental', 'timescaledb_information', '_timescaledb_functions',
+                                'information_schema', 'pg_catalog') AND
+                            d.datname = current_database()
+                        group by xact_commit;
+                end;
+
+                with before as (
+                    select * from _mig_eval_t where n = 1
+                ), after as (
+                    select * from _mig_eval_t where n = 2
+                )
+                select
+                    round((after.n_tup_ins - before.n_tup_ins) / @wait@, 3) inserts_per_sec,
+                    round((after.n_tup_upd - before.n_tup_upd) / @wait@, 3) updates_per_sec,
+                    round((after.n_tup_del - before.n_tup_del) / @wait@, 3) deletes_per_sec,
+                    round((after.xact_commit - before.xact_commit) / @wait@, 3) txns_per_sec
+                from after, before;
+
+                drop table _mig_eval_t cascade;
+            commit;
+            """,
+    }, {
+        "name": "Do compressed chunks have mutable (insert/update/delete) compression*",
+        "query": f"""
+            select exists(
+                select 1 from _timescaledb_catalog.chunk
+                where
+                    status = 9 and
+                    compressed_chunk_id is not null
```

Comment:
I _think_ that we actually only care about UPDATE/DELETE statements on rows in compressed chunks (because we will handle the replication of inserts to compressed chunks correctly).

This checks whether the chunk is marked as partial, which can also (and probably will mostly) be caused by INSERTs into the compressed chunk.

That is to say, I don't have confidence that this question actually tells us something useful.

Reply:
Since this does not completely answer our question of mutable compression and the point that the information might not be useful, I have removed this query in the latest commits.